parser.py

```diff
@@ -12,19 +13,27 @@
 def p_process(p):
     '''
     process : create EXCLAMATION create EXCLAMATION join EXCLAMATION communication
+            | join EXCLAMATION communication
     '''
 
+def p_create_DEFAULT(p):
+    '''
+    create : DEFAULT LB ID RB
+    '''
+    controller.create_server(p[3],socket.gethostbyname(socket.gethostname()),random.randint(2000, 50000))
```

Comment:
We should try and keep everything socket related out of the parser file. How about putting an if statement in the create_server function specify what type of server we want to create from the parser.

Reply:
I reverted to original, but now the function calls create server instead.